v22.2/trigram-indexes.md

```diff
@@ -0,0 +1,360 @@
+---
+title: Trigram Indexes
+summary: Trigram indexes on STRING columns improve the performance of fuzzy string matching and pattern matching in large datasets.
+toc: true
+keywords: gin, gin index, gin indexes, inverted index, inverted indexes, accelerated index, accelerated indexes, trigram, trigrams
+docs_area: develop
+---
+
+{% include_cached new-in.html version="v22.2" %} A _trigram index_ is a type of [inverted index](inverted-indexes.html) created on a [`STRING`](string.html) column. Trigram indexes are used to efficiently search for strings in large tables without providing an exact search term.
+
+This page describes how to create and use trigram indexes on CockroachDB.
+
+{{site.data.alerts.callout_info}}
+Some PostgreSQL syntax and features are currently unsupported. For details, see [Unsupported features](#unsupported-features).
+{{site.data.alerts.end}}
+
+## How do trigram indexes work?
+
+Trigram indexes make [substring and similarity matches](https://www.postgresql.org/docs/current/pgtrgm.html) efficient by indexing the unique trigrams of a string. A trigram is a group of three consecutive characters (including spaces) in a string.
```

Comment:
Just noticed that there's another slight flaw here - actually, trigrams don't include spaces or whitespace. It looks like they do, because the trigrams at the beginning and end of a word (words are produced by splitting the input on whitespace, and discarding the whitespace) include artificially inserted spaces to help distinguish the starts and ends of words.

tl;dr You may not need to include this detail but trigrams don't really include the whitespace in strings.

Reply:
Thanks for correcting this! Fixed.